src/llm_utils.py

```diff
@@ -0,0 +1,182 @@
+
+import os
+import json
+from google.genai import types
+from google import genai
+from typing import List, Dict
+
+def analyze_code_changes(diff_content: str) -> List[Dict]:
+    """
+    Analyze code changes using OpenAI's GPT model
+    Returns a list of review comments
+    """
+    API_KEY = os.getenv('GEMINI_API_KEY')
+
+    if not API_KEY:
+        raise ValueError("GEMINI_API_KEY is not set. Please add it to your environment variables.")
+
+    # Prepare the prompt for the LLM
+    prompt = f"""
+    Analyze the following code changes and provide review comments using EXACTLY this format:
+
+    FILE: [file-path]
+    LINE: [line-number]
+    COMMENT: [your comment]
+    SUGGESTION: [optional suggested code]
+
+    ---
+    Example:
+    FILE: src/app.py
+    LINE: 42
+    COMMENT: Avoid magic numbers, consider using a constant
+    SUGGESTION: MAX_RETRIES = 3
+
+    ---
+
+    Make sure to insert '---' in between two review comments. IT IS A MUST!!!
+
+    Focus on:
+    - Code quality issues
+    - Security vulnerabilities
+    - Performance optimizations
+    - Style inconsistencies
+    - Possible bugs
+    - Missing documentation
+
+    Diff content:
+    {diff_content}
+    """
+
+    client = genai.Client(api_key=API_KEY)
+
+    #Get analysis from Gemini model
+    response = client.models.generate_content(
+        model="gemini-2.0-flash", contents=prompt
+    )   
+
+    #print(response.text)
+    
+    # Parse and format the response
+    review_comments = parse_llm_response(response.text)
+    return review_comments
+
+
+
+"""
+This is the final code given by deepseek to parse the response from the LLM model. It includes errors
+"""
+
+def parse_llm_response(response: str) -> List[Dict]:
+    review_comments = []
+    
+    # Split response into valid comment blocks
+    comment_blocks = response.split('---')
+    
+    for block in comment_blocks:
+        block = block.strip()
+        if not block:
+            continue
+            
+        comment_data = {
+            'path': None,
+            'line': None,
+            'body': ''
```

Comment:
The comment suggests this is the final code from DeepSeek, but mentions it includes errors. It would be beneficial to identify and address these errors for improved code quality. Consider using more descriptive variable names.

```suggestion
review_comments = parse_llm_response(response.text)
```